atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepQueueDeleter.java

```diff
@@ -47,14 +47,21 @@ public class SweepQueueDeleter {
     /**
      * Executes targeted sweep, by inserting ranged tombstones corresponding to the given writes, using the sweep
      * strategy determined by the sweeper.
-     *
-     * @param unfilteredWrites individual writes to sweep for. Depending on the strategy, we will insert a ranged
+     *  @param unfilteredWrites individual writes to sweep for. Depending on the strategy, we will insert a ranged
      * tombstone for each write at either the write's timestamp - 1, or at its timestamp.
-     * @param sweeper supplies the strategy-specific behaviour: the timestamp for the tombstone and whether we must use
-     * sentinels or not.
+     * @param shardAndStrategy is used to create a sweeper that supplies the strategy-specific behaviour: the timestamp
+     *                         for the tombstone and whether we must use
      */
-    public void sweep(Collection<WriteInfo> unfilteredWrites, Sweeper sweeper) {
+    public void sweep(Collection<WriteInfo> unfilteredWrites, ShardAndStrategy shardAndStrategy) {
+        // todo(snanda): we still need to abort and populate A :(
+        if (shardAndStrategy.isNoop()) {
```

Comment:
Basically, by this time the batch to be swept has already been read and transactions been concluded. We want no op for writes on tables that are not being swept.